### PR TITLE
sanitize javascript: urls for <object> tags

### PR DESCRIPTION
**Pull request by kassens**
sanitize javascript: urls for <object> tags

React 19 added sanitization for `javascript:` URLs for `href` properties on various tags. This PR also adds that sanitization for `<object>` tags as well that Firefox otherwise executes.

### CODE REVIEW

File: packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js [hunk 1]
```diff
@@ -1202,6 +1202,7 @@ function pushAttribute(
       pushStyleAttribute(target, value);
       return;
     }
+    case 'data':
```
Comment (sebmarkbage) [inline packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js:1205]: I'd just follow `pushStartAnchor`'s lead and make a specific `pushStartObject` method that does the sanitization specifically for that tag.

**Comment by kassens** [comment]
My open question is if we have preferred behavior of `<object data="" />`

**Comment by sebmarkbage** [comment]
What does `<object data="">` do in the browser?

- If it does something useful, then we shouldn't error.
- If it loads the current page's URL as an external object, then we should error.

The goal is to not accidentally load current page URL a second time for no good reason.

**Comment by sebmarkbage** [comment]
Seems like you have some failing tests too.

**Comment by kassens** [comment]
Tests pass now.

Implemented removing the string removal to match general empty string behavior.

I couldn't repro the behavior of re-requesting the page either with imgs or object tags, maybe I'm missing something or it doesn't show up in the network panel. Had a repro in this sandbox: https://codesandbox.io/p/sandbox/upbeat-hofstadter-dvmlwv?file=%2Findex.html%3A12%2C1

**Comment by eps1lon** [comment]
> I couldn't repro the behavior of re-requesting the page either with imgs or object tags, maybe I'm missing something or it doesn't show up in the network panel.

It seems like Chrome (and possibly other browsers?) are no longer trying to render an image with an empty `src` attribute. Apparently this is even specced behavior: https://github.com/whatwg/html/issues/3280#issuecomment-351225693

**Review by sebmarkbage — APPROVED** [review]

**Pull request merged**